model_compression_toolkit/hardware_model/framework_hardware_model.py

```diff
@@ -0,0 +1,128 @@
+from model_compression_toolkit.hardware_model.operators import OperatorsSet
+
+
+from model_compression_toolkit.hardware_model.hardware_model import \
+    HardwareModel
+from model_compression_toolkit.hardware_model.current_framework_hardware_model import _current_framework_hardware_model
+
+from model_compression_toolkit.hardware_model.framework_hardware_model_component import \
+    FrameworkHardwareModelComponent
+from model_compression_toolkit.hardware_model.layer_filter_params import \
+    LayerFilterParams
+from model_compression_toolkit.hardware_model.operations_to_layers import OperationsToLayers, OperationsSetToLayers
+
+
+
+
+class FrameworkHardwareModel(ImmutableClass):
+    def __init__(self,
+                 hw_model: HardwareModel,
+                 name: str="base"):
+
+        super().__init__()
+        self.name = name
+        assert isinstance(hw_model, HardwareModel)
+        self.hw_model = hw_model
+        self.op_sets_to_layers = OperationsToLayers()
+        self.layer2qco, self.filterlayer2qco = {}, {}
+        self.hwm_opsets_not_used = [s.name for s in hw_model.operator_set]
+        self.remove_fusing_names_from_not_used_list()
+
+    def get_op_layers(self, op: OperatorsSet):
+        return self.op_sets_to_layers.get_layers_by_op(op)
+
+    def get_fusing_patterns(self):
+        res = []
+        for p in self.hw_model.fusing_patterns:
+            ops = [self.get_op_layers(x) for x in p.operator_groups_list]
+            res.extend(itertools.product(*ops))
+        return [list(x) for x in res]
+
+
+    def get_info(self):
+        return {"Framework Hardware Model": self.name,
+                "Hardware model": self.hw_model.get_info(),
+                "Operations to layers": {op2layer.name:[l.__name__ for l in op2layer.layers] for op2layer in self.op_sets_to_layers.op_sets_to_layers}}
+
+    def show(self):
+        pprint.pprint(self.get_info(), sort_dicts=False, width=110)
+
+
+
+
+    def append_component(self, hm_component: FrameworkHardwareModelComponent):
+        if isinstance(hm_component, OperationsSetToLayers):
+            self.op_sets_to_layers += hm_component
+        else:
+            raise Exception(f'Trying to append an unfamiliar HardwareModelComponent of type: {type(hm_component)}')
+
+    def __enter__(self):
+        _current_framework_hardware_model.set(self)
+        return self
+
+    def __exit__(self, exc_type, exc_value, tb):
+        if exc_value is not None:
+            print(exc_value, exc_value.args)
+            raise exc_value
+        self.raise_warnings()
+        self.layer2qco, self.filterlayer2qco = self._get_config_options_mapping()
+        _current_framework_hardware_model.reset()
+        self.initialized_done()
+        return self
+
+
+    def get_qco_by_layer(self, node: BaseNode)->QuantizationConfigOptions:
+        # For now, just use default qc options for all nodes.
+        return self.hw_model.default_qco
```

Comment:
Maybe we should add a function to get default_qco? (and use it instead of get_qco_by_layer)